### DISCUSSION TITLE
How to run tobac with my own 2D data

### DISCUSSION THREAD

**Discussion by zhangzh1118** [tobac-project/tobac#228] [Q&A]
Hey, I am using the output data from A two-dimensional (x-z) slab-symmetric cloud model, which has a vertical and horizontal resolution of 300m and 100m respectively. The grids of the model does not have a corresponding latitude and longitude.

I want to detect clouds using a constraint that the  drop mixing ratio is greater than 0.01 kg/kg in the vertical profile plot.

I have the data in txt files and want to write them into nc files to run tobac. l would like to ask which variables in nc files are necessary for tobac to run. thank you for your time to read my question!

**Comment by freemansw1** [comment]
You should be able to get *tobac* to track clouds in any dataset that you can put into Iris (https://scitools-iris.readthedocs.io/en/latest/) format. I recommend loading your data into an xarray (https://docs.xarray.dev/en/stable/) DataArray first, then converting that to Iris. tobac doesn't require any variables to be present other than `time`. 

You will run into issues with tracking; *tobac* does not currently have support for tracking with unequal grid spacing; it will need to assume that your grid is equally spaced.

**Comment by freemansw1** [comment]
@zhangzh1118 did you end up deciding to use *tobac*? If so, has it worked for you?
